boards/arm/secureiot1702/doc/secureiot1702.rst

```diff
@@ -0,0 +1,82 @@
+.. _secureiot1702:
+
+Microchip SecureIoT1702
+#######################
+
+Overview
+********
+
+Board configuration for Microchip CEC1702 demo board. This is a cryptographic
+embedded controlled based on an ARM Cortex-M4.
+
+Highlights of the board:
+
+- CEC1702 32-bit ARM® Cortex®-M4F Controller with Integrated Crypto
+- Compact, high-contrast, serial graphic LCD Display Module with back-light
+- 2x4 matrix of push buttons inputs
+- USB-UART Converter as debug interface
+- Potentiometer to ADC channel
+- Serial Quad I/O (SQI) flash
+- OTP programmable in CEC1702
+- Two expansion headers compatible with MikroElektronika mikroBUS™ Expansion interface
+
+More information can be found from `SecureIoT1702 website`_ and
+`CEC1702 website`_. The SoC programming information is available
+in `CEC1702 datasheet`_.
+
+Supported Features
+==================
+
+Following devices are supported:
+
+- Nested Vectored Interrupt Controller (NVIC)
+- System Tick System Clock (SYSTICK)
+- Serial Ports (NS16550)
+- Quad Master-only SPI controller (QMSPI)
+
+
+Connections and IOs
+===================
+
+Please refer to the `SecureIoT1702 schematics`_ for the pin routings.
+Additional devices can be connected via mikroBUS expansion interface.
+
+Programming and Debugging
+*************************
+
+Flashing
+========
+
+The produced zephyr.bin needs to be converted to bootable image using
+the CEC1702 SPI Image Generator Utility. For more information see
+the `_CEC1702 quick start guide`_.
+
+This image can then be programmed to the SPI flash using flashrom,
+or MikroProg for CEC.
+
+Debugging
+=========
+
+The MCU has standard JTAG debugging features. But the author of this port
```

Comment:
Delete "But the author of this port has not used it."  Can you provide a reference link about this in the microchip documentation you're linking to?

Reply:
@franciscomunoz Seems you added standard Debugging example for the MEC boards. Does that work? If yes, it should work also for CEC and I could just add the same here.

Reply:
@fabled  It works on the MEC1701 and MEC1501. It could work on the CEC1702 for sure, but very basic features as it may require extra support from the tool vendor.

Reply:
Ok. Maybe I copy that over and state is as "should" work. I can also look into cloning the parts of the Building steps. But the `objcopy` step is not needed for me. I also have python based tool that can do the `zephyr.bin` conversion to SPI flashable image, but I think that is worth a separate commit. @franciscomunoz Do you know if MEC is using same SPI image format as CEC? I think it does, it's mostly the same thing. I also saw you got the gpio and pinmux stuff done, those could be reused in CEC side too.

Reply:
@fabled It worked on the 1701, You just need to update the base address. Another thing I remember is the script ran with python2.7. It never ran with python3.X for me. Right now there are gpio,pimux, i2c, eSPI(subsystem is being reviewed). More stuff is being cooked. Drivers should work on the CEC (in fact this is why there is an Xec). However, a new HAL is going to be needed for the CEC to abstract the registers in the same way it is done for the MEC1501.

Reply:
> @fabled However, a new HAL is going to be needed for the CEC to abstract the registers in the same way it is done for the MEC1501.
I am thinking it might be better to try to avoid Microchip HAL. Similar how the UART code is shared, and use DT for providing all the base addresses, interrupts etc. Perhaps the gpio,pinmux,etc. could be made to have the register offsets internally, and take everything else from DT and be independent from the HAL. What do you think?
Also does MEC have the same QMSPI? If yes, should that be renamed to xec_qmspi and also used in the MEC code?

Reply:
@fabled I can't make any of this decisions as there are several people involved. The easiest path for would be to have the CEC1702 HAL.